src/lib/sticky-header/sticky-header.ts

```diff
@@ -0,0 +1,299 @@
+
+    // Have to add the translate3d function for the sticky element's css style.
+    // Because iPhone and iPad's browser is using its owning rendering engine. And
+    // even if you are using Chrome on an iPhone, you are just using Safari with
+    // a Chrome skin around it.
+    //
+    // Safari on iPad and Safari on iPhone do not have resizable windows.
+    // In Safari on iPhone and iPad, the window size is set to the size of
+    // the screen (minus Safari user interface controls), and cannot be changed
+    // by the user. To move around a webpage, the user changes the zoom level and position
+    // of the viewport as they double tap or pinch to zoom in or out, or by touching
+    // and dragging to pan the page. As a user changes the zoom level and position of the
+    // viewport they are doing so within a viewable content area of fixed size
+    // (that is, the window). This means that webpage elements that have their position
+    // "fixed" to the viewport can end up outside the viewable content area, offscreen.
+    //
+    // So the 'position: fixed' does not work on iPhone and iPad. To make it work,
+    // 'translate3d(0,0,0)' needs to be used to force Safari re-rendering the sticky element.
+    this.element.style.transform = 'translate3d(0px,0px,0px)';
+
+    let stuckRight: any = this.upperScrollableContainer.getBoundingClientRect().right;
+
+    let stickyCss = {
+      position: 'fixed',
+      top: this.upperScrollableContainer.offsetTop + 'px',
+      right: stuckRight + 'px',
+      left: this.upperScrollableContainer.offsetLeft + 'px',
+      bottom: 'auto',
+      width: this._originalStyles.width,
+      zIndex: this.zIndex + '',};
+    extendObject(this.element.style, stickyCss);
+  }
+
+  /**
+   * Unstuck element: When an element reaches the bottom of its cdkStickyRegion,
+   * It should be unstuck. And its position will be set as 'relative', its bottom
+   * will be set as '0'. So it will be stick at the bottom of its cdkStickyRegion and
+   * will be scrolled up with its cdkStickyRegion element. In this way, the sticky header
+   * can be changed smoothly when two sticky header meet and the later one need to replace
+   * the former one.
+   */
+  private _unstuckElement(): void {
+    this.isStuck = false;
+
+    if (!this.stickyParent) {
+      return;
+    }
+
+    this.element.classList.add(STICK_END_CLASS);
+    this.stickyParent.style.position = 'relative';
+    let unstuckCss = {
+      position: 'absolute',
+      top: 'auto',
+      right: '0',
+      left: 'auto',
+      bottom: '0',
+      width: this._originalStyles.width};
+    extendObject(this.element.style, unstuckCss);
+  }
+
+
+  /**
+   * 'sticker()' function contains the main logic of sticky-header. It decides when
+   * a header should be stick and when should it be unstuck by comparing the offsetTop
+   * of scrollable container with the top and bottom of the sticky region.
+   */
+  sticker(): void {
```

Comment:
How about something like `_applyStickyPositionStyles`? The name `sticker` doesn't tell you much about what the function does.